src/geode/inspector/criterion/intersections/surface_intersections.cpp

```diff
@@ -0,0 +1,404 @@
+                { t1.vertices()[v_id], t1.vertices()[v2_id] }, t2 );
+            if( intersection.first != geode::Position::outside )
+            {
+                if( common_points.size() != 1 )
+                {
+                    return true;
+                }
+                const auto edge_inter_pt_id =
+                    vertex_position_to_index( intersection.first );
+                const auto t2_inter_pt_id =
+                    vertex_position_to_index( intersection.second );
+                if( edge_inter_pt_id == geode::NO_LID
+                    || t2_inter_pt_id == geode::NO_LID )
+                {
+                    return true;
+                }
+                if( t1_vertices[edge_inter_pt_id == 0 ? v_id : v2_id]
+                        != t2_vertices[t2_inter_pt_id]
+                    || t2_vertices[t2_inter_pt_id] != common_points[0] )
+                {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    template <>
+    bool TriangleTriangleIntersection< 3 >::triangles_intersect(
+        geode::index_t t1_id,
+        geode::index_t t2_id,
+        absl::Span< const geode::index_t > common_points ) const
+    {
+        const auto t1_vertices = mesh_.polygon_vertices( t1_id );
+        const auto t2 = mesh_.triangle( t2_id );
+        if( common_points.size() == 2 )
+        {
+            const auto& t1_third_pt =
+                mesh_.point( third_point_index( t1_vertices, common_points ) );
+            if( geode::segment_triangle_intersection_detection(
+                    { mesh_.point( common_points[0] ), t1_third_pt }, t2 )
+                        .first
+                    == geode::Position::parallel
+                || geode::segment_triangle_intersection_detection(
+                       { mesh_.point( common_points[1] ), t1_third_pt }, t2 )
+                           .first
+                       == geode::Position::parallel )
+            {
+                return true;
+            }
+            return false;
+        }
+        const auto t1 = mesh_.triangle( t1_id );
+        const auto t2_vertices = mesh_.polygon_vertices( t2_id );
+        if( triangle_intersects_other(
+                t1, t2, t1_vertices, t2_vertices, common_points )
+            || triangle_intersects_other(
+                t1, t2, t1_vertices, t2_vertices, common_points ) )
+        {
+            return true;
+        }
+        return false;
+    }
+} // namespace
+
+namespace geode
+{
+    template < index_t dimension >
+    class SurfaceMeshIntersections< dimension >::Impl
+    {
+    public:
+        Impl( const SurfaceMesh< dimension >& mesh, bool verbose )
+            : mesh_( dynamic_cast< const TriangulatedSurface< dimension >& >(
+                mesh ) ),
+              verbose_( verbose )
+        {
+        }
+
+        bool mesh_has_auto_intersections() const
```

Comment:
```suggestion
        bool mesh_has_self_intersections() const
```
Compiling with Arnaud's suggestion